docs for workflow payload schema (#875)
novuhq/docs: content/docs/platform/workflow/build-a-workflow.mdx

Schema validation can be enabled per workflow. When active, Novu validates incoming payloads against the schema when the workflow is triggered.

This means:
- Missing required properties will cause the request to fail.
- Data types must match exactly (for example, a string cannot be passed where a number is expected).
- Invalid values are rejected before the workflow executes.

This validation is applied at the HTTP trigger level, and prevents invalid data from entering the workflow entirely.

### Schema configuration options

When defining a schema property, the available configuration fields vary depending on the selected type.

#### General fields (for all types)

- Property name
- Property type
- Required property checkbox
- Default value – Optional fallback if no value is provided
- Min length and max length

#### Type-specific configuration

Depending on the selected type, additional configuration options appear:

- String:
  - Format: None, date-time, date, time, duration, email, hostname, ipv4, ipv6, uuid, uri-reference, uri-template, json-pointer, relative-json-pointer, regex
  - Pattern: Regex-based validation

> - Enum: You’ll be able to add choices, a list of predefined, allowed values. This restricts the field to only those values.

Reviewer: ```suggestion
- Enum: Add choices, which are a list of predefined, allowed values. This restricts the field to only those values.
```

> - Array: You’ll be able to select the Array item type, this defines the data type of each array element.

Reviewer: ```suggestion
- Array: Select the Array item type, which defines the data type of each array element.
```

> - Object: You’ll be able to add nested properties, each with their own type, required status, and validation options.

Reviewer: ```suggestion
- Object: Add nested properties, each with their own type, required status, and validation options.
```
